Source/ITab_Lock.cs

```diff
@@ -141,32 +143,31 @@ protected override void FillTab()
             }
             if (Clipboard.StoredState.HasValue && Widgets.ButtonText(pasteButtonsRect, "Locks_Paste".Translate()))
             {
-                Data.WantedState.CopyFrom(Clipboard.StoredState.Value);
+                SetWantedStateData(SelDoor, Clipboard.StoredState.Value);
                 OnOpen();
             }
 
             //TODO
             //PlayerKnowledgeDatabase.KnowledgeDemonstrated(ConceptDefOf.StorageTab, KnowledgeAmount.FrameDisplayed);
             GUI.EndGroup();
 
-            UpdateSettings();
+            UpdateSettings(anythingChanged);
         }
 
-        private void UpdateSettings()
+        private void UpdateSettings(bool anythingChanged)
         {
             if (SelDoor == null)
                 return;
             if (lastSelectedThing == SelThing)
             {
-                Data.WantedState.locked = locked;
-                Data.WantedState.mode = vistitorsAllowed ? LockMode.Allies : LockMode.Colony;
-                Data.WantedState.petDoor = petDoor;
-                Data.WantedState.pensDoor = pensDoor;
-                Data.WantedState.allowSlave = slaveDoor;
-                Data.WantedState.allowAnimals = animalsAllowed;
-                Data.WantedState.childLock = childLock;
-                if (Data.NeedChange)
-                    LockUtility.UpdateLockDesignation(SelDoor);
+                if (anythingChanged)
```

Comment:
Any particular reason why not use an existing solution for tracking changes?

Reply:
I'm assuming you mean `LockData.NeedChange`, right? That's primarily because of how syncing works with Multiplayer.

To use `LockData.NeedChange`, the values in the object would need to be changed. If those values were set instantly on the side of the player who applied those changes, it will end in a situation where those values are different between players (until they are synced). And since those values are different for multiple people, this causes issues.

I work around setting the `anythingChanged` to true. I could alternatively compare the current `WantedState` with the values in the ITab, and apply them if there's anything that was changed. Or I could make a new, temporary `LockData` object and use that instead, but this approach seems less than ideal.

Reply:
Make sense, thanks for explanation